crates/ironrdp-pdu/src/basic_output/fast_path.rs

```diff
@@ -346,14 +365,25 @@ impl From<&FastPathUpdate<'_>> for UpdateCode {
     }
 }
 
-#[derive(Debug, Copy, Clone, PartialEq, Eq, FromPrimitive, ToPrimitive)]
+#[derive(Debug, Copy, Clone, PartialEq, Eq, FromPrimitive)]
 pub enum Fragmentation {
     Single = 0x0,
     Last = 0x1,
     First = 0x2,
     Next = 0x3,
 }
 
+impl Fragmentation {
+    pub fn as_u8(&self) -> u8 {
+        match self {
+            Self::Single => 0x0,
+            Self::Last => 0x1,
+            Self::First => 0x2,
+            Self::Next => 0x3,
+        }
+    }
```

Comment:
suggestion: This should do the trick!

```suggestion
    pub fn as_u8(self) -> u8 {
        self as u8
    }
```

Reply:
On the other hand, when we add a lint for `as` conversion, we will need to mark all these functions as `#[expect(clippy::as_conversions)]`. Wouldn't it be better to use `match` as the middle ground? I understand that it's tedious, but it won't require a lint silence.

Reply:
Using `#[expect(clippy::as_conversions)]` for this specific kind of conversion is fine :+1:
For enum -> integer, `as` performs a value-preserving conversion of the discriminant.
Just make sure there is `#[repr(u8)]` when casting to `u8`, so it’s total, safe, and cannot produce an invalid value.
Indeed, there _is_ a pitfall, which is why we have this warning: silent narrowing, sign changes _if_ you cast to a different size/signedness (e.g., `#[repr(u16)]` but `as u8`). But if we encapsulate the logic at one place, like here, in a `as_u8` function, it’s perfectly satisfying :+1: 

Also add a reason such as `#[repr(u8)] guarantees discriminant layout, and `as` is the only way to cast enum -> primitive`